tools/dev-doctor/check.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+// A checkResult is the result of a check.
+type checkResult int
```

Comment:
Nit: I find "check" to be slightly redundant. This variable and all references following in this file can probably drop "check" from them, and it'll still be clear enough.

Reply:
Personally, I prefer the inclusion of `check` in the types as it links the concepts together. `binaryCheck`, `unameCheck`, and `checkResult` are clearly related, whereas `binary`, `uname`, and `result` are not.